### PR TITLE
Sonar Opening Angles as Properties

### PR DESCRIPTION
**Pull request by romulogcerqueira**
As properties, we can simulate both gemini and blueview sonars using this same component.

### CODE REVIEW

File: imaging_sonar_simulation.orogen [hunk 1]
```diff
@@ -29,6 +29,12 @@ task_context "Task" do
     # Number of Bins
     property("number_of_bins", "int", 500).dynamic
 
+    # Opening of the beam orthogonal to the device's Z direction
+    property("beam_width", "double")
```
Comment (doudou) [inline imaging_sonar_simulation.orogen]: Why not a base::Angle right away ?

Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: I thought the same as you, however I need to change another scanning sonar properties too (`start_angle`, `end_angle` and `step_angle`).
I toke your comment and added a new commit with these changes.

Comment (romulogcerqueira) [inline imaging_sonar_simulation.orogen]: @doudou, I discovered a corner case problem on codeSonarData() method from the driver. 
Then this pull request depends on https://github.com/Brazilian-Institute-of-Robotics/simulation-gpu_sonar_simulation/pull/11.

File: imaging_sonar_simulation.orogen [hunk 2]
```diff
@@ -57,7 +64,10 @@ task_context "MultibeamSonarTask" do
     subclasses "imaging_sonar_simulation::Task"
 
     # Number of Beams
-    property("number_of_beams", "int", 256)
+    property("beam_count", "int", 256)
     
+    # Multibeam sonar orientation
+    property("orientation", "imaging_sonar_simulation/orientation/Type", :Horizontal).dynamic
```
Comment (doudou) [inline imaging_sonar_simulation.orogen]: Isn't the sonar pose given to the task ? Get rid of this, and make sure that the pose can be rotated (i.e. with Gazebo)

File: tasks/Task.cpp [hunk 3]
```diff
@@ -38,19 +38,11 @@ void Task::cleanupHook() {
...
-	vizkit3dWorld->getWidget()->setAxes(false);
-	vizkit3dWorld->getWidget()->setAxesLabels(false);
+void Task::init(const base::Angle& fovX, const base::Angle& fovY, uint value, float range, bool isHeight) {
 
 	// init shader
-	float radX = degX * (M_PI / 180.0) * 0.5;
-	float radY = degY * (M_PI / 180.0) * 0.5;
-	_normal_depth_map = vizkit3d_normal_depth_map::NormalDepthMap(range, radX, radY);
-	_capture = vizkit3d_normal_depth_map::ImageViewerCaptureTool(degY, degX, value, isHeight);
+	_normal_depth_map = vizkit3d_normal_depth_map::NormalDepthMap(range, fovX.rad * 0.5, fovY.rad * 0.5);
```
Comment (doudou) [inline tasks/Task.cpp]: *Half* the field of view ?

Comment (romulogcerqueira) [inline tasks/Task.cpp]: The `NormalDepthMap` constructor requires the half-FOV.

In the old code, I pass the same half-FOV as `radX` and `radY`. I excluded the variables, but the result is the same.

File: imaging_sonar_simulation.orogen (hunk 2 above)
Comment (doudou) [inline imaging_sonar_simulation.orogen]: https://trello.com/c/GwGSA36N

**Pull request merged**